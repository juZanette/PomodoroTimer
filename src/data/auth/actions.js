import { firebase } from '../../firebase/firebase'

export const setUserInfo = (userInfo) => ({
  type: 'SET_USER_INFO',
  userInfo,
})

export const linkAccount = (authProvider) => {
  return async (dispatch) => {
    try {
      const res = await firebase.auth().currentUser.linkWithPopup(authProvider)

      const user = res.user

      const userInfo = { uid: user.uid }

      userInfo.creationTime = user.metadata.creationTime

      if (user.providerData && user.providerData.length) {
        userInfo.name = user.providerData[0].displayName
        userInfo.photo = user.providerData[0].photoURL
      }

      dispatch(setUserInfo(userInfo))
    } catch (e) {
      if (e.code === 'auth/credential-already-in-use') {
        await firebase.auth().signInWithCredential(e.credential)
      }
    }
  }
}

export const setCreationTime = (creationTime) => ({
  type: 'SET_CREATION_TIME',
  creationTime,
})
