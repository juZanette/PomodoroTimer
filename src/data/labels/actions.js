import fs from '../../firebase/firebase'

export const setFullscreenDialog = (fullscreenDialog) => ({
  type: 'SET_FULLSCREEN_DIALOG',
  fullscreenDialog,
})

export const setDesktopDialog = (desktopDialog) => ({
  type: 'SET_DESKTOP_DIALOG',
  desktopDialog,
})

export const setDeleteAlert = (deleteAlert) => ({
  type: 'SET_DELETE_ALERT',
  deleteAlert,
})

export const setMenuOpened = (menuOpened) => ({
  type: 'SET_MENU_OPENED',
  menuOpened,
})

export const setFormValue = (formValue) => ({
  type: 'SET_FORM_VALUE',
  formValue,
})

export const setLabelEditting = (labelEditting) => ({
  type: 'SET_LABEL_EDITTING',
  labelEditting,
})

export const setLabelSelected = (labelSelected) => ({
  type: 'SET_LABEL_SELECTED',
  labelSelected,
})

export const addLabel = (label) => ({
  type: 'ADD_LABEL',
  label,
})

export const startAddLabel = (label) => {
  return async (dispatch, getState) => {
    const uid = getState().auth.uid

    const ref = await fs.collection(`users/${uid}/labels`).add(label)

    dispatch(
      addLabel({
        id: ref.id,
        ...label,
      })
    )

    return ref
  }
}

export const editLabel = (id, updates) => ({
  type: 'EDIT_LABEL',
  id,
  updates,
})

export const startEditLabel = (id, updates) => {
  return async (dispatch, getState) => {
    const uid = getState().auth.uid

    await fs.doc(`users/${uid}/labels/${id}`).update({
      ...updates,
    })

    dispatch(editLabel(id, updates))
  }
}

export const deleteLabel = (id) => ({
  type: 'DELETE_LABEL',
  id,
})

export const startDeleteLabel = (id) => {
  return async (dispatch, getState) => {
    const uid = getState().auth.uid
    await fs.doc(`users/${uid}/labels/${id}`).delete()

    dispatch(deleteLabel(id))
  }
}

export const setLabels = (labels) => ({
  type: 'SET_LABELS',
  labels,
})

export const startSetLabels = () => {
  return async (dispatch, getState) => {
    const uid = getState().auth.uid

    const labelsRef = await fs.collection(`users/${uid}/labels`).get()

    const labels = labelsRef.docs.map((label) => ({
      id: label.id,
      ...label.data(),
    }))

    dispatch(setLabels(labels))

    return labelsRef
  }
}
