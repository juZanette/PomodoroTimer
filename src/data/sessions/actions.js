import fs from '../../firebase/firebase'

export const setSessions = (sessions) => ({
  type: 'SET_SESSIONS',
  sessions,
})

export const startSetSessions = () => {
  return async (dispatch, getState) => {
    const uid = getState().auth.uid

    const sessionsRef = await fs.collection(`users/${uid}/sessions`).get()

    const sessions = sessionsRef.docs.map((session) => ({
      id: session.id,
      ...session.data(),
    }))

    dispatch(setSessions(sessions))

    return sessionsRef
  }
}

export const addSession = (session) => ({
  type: 'ADD_SESSION',
  session,
})

export const startAddSession = (session) => {
  return async (dispatch, getState) => {
    const uid = getState().auth.uid
    const { duration, label = null, createdAt } = session

    const ref = await fs.collection(`users/${uid}/sessions`).add({
      duration,
      label,
      createdAt,
    })

    dispatch(
      addSession({
        id: ref.id,
        ...session,
      })
    )

    return ref
  }
}
