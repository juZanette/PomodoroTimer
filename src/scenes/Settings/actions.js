export const setRounds = roundsNumber => ({
  type: 'SET_ROUNDS',
  roundsNumber,
})
