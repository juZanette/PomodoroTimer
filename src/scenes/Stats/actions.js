export const setRounds = () => ({
  type: 'INCREMENT_TOTAL_SESSIONS',
})
