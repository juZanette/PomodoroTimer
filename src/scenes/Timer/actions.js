export const setStatus = status => ({
  type: 'SET_STATUS',
  status,
})

export const setProgress = progress => ({
  type: 'SET_PROGRESS',
  progress,
})

export const setTimeLeft = timeLeft => ({
  type: 'SET_TIME_LEFT',
  timeLeft,
})

export const saveInterval = interval => ({
  type: 'SAVE_INTERVAL',
  interval,
})

export const pauseTimer = () => ({
  type: 'PAUSE_TIMER',
})

export const resetTimer = () => ({
  type: 'RESET_TIMER',
})

export const setNextTimer = () => ({
  type: 'SET_NEXT_TIMER',
})
