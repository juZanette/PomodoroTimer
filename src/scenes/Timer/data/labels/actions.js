export const setDialogOpened = (dialogOpened) => ({
  type: 'SET_DIALOG_OPENED',
  dialogOpened,
})

export const setAlertOpened = (alertOpened) => ({
  type: 'SET_ALERT_OPENED',
  alertOpened,
})

export const setLabelToEdit = (labelToEdit) => ({
  type: 'SET_LABEL_TO_EDIT',
  labelToEdit,
})

export const setFormValue = (formValue) => ({
  type: 'SET_FORM_VALUE',
  formValue,
})
