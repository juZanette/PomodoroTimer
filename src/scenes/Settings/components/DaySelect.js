import React, { useEffect, useState } from 'react'
import { useSelector, useDispatch } from 'react-redux'
import styled from 'styled-components'
import Box from '@material-ui/core/Box'
import MatFormControl from '@material-ui/core/FormControl'
import Select from '@material-ui/core/Select'
import InputLabel from '@material-ui/core/InputLabel'
import MenuItem from '@material-ui/core/MenuItem'
import { startSetFirstDayOfTheWeek } from '../../../data/settings/actions'

export const DaySelect = () => {
  const [day, setDay] = useState('')

  const firstDayOfTheWeek = useSelector(
    (state) => state.settings.firstDayOfTheWeek
  )

  useEffect(() => {
    if (firstDayOfTheWeek !== null && firstDayOfTheWeek !== undefined) {
      setDay(firstDayOfTheWeek)
    }
  }, [firstDayOfTheWeek])

  const dispatch = useDispatch()

  const handleChange = (event) => {
    const day = event.target.value

    setDay(day)
    dispatch(startSetFirstDayOfTheWeek(day))
  }

  return (
    <Box my={2}>
      <FormControl variant="outlined">
        <InputLabel id="day-select">Week starts on</InputLabel>
        <Select
          labelId="day-select"
          value={day}
          onChange={handleChange}
          label="Week starts on"
        >
          <MenuItem value="Monday">Monday</MenuItem>
          <MenuItem value="Sunday">Sunday</MenuItem>
        </Select>
      </FormControl>
    </Box>
  )
}

const FormControl = styled(MatFormControl)`
  width: 100%;
`
