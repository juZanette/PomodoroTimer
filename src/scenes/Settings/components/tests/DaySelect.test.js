import React from 'react'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { createMount } from '@material-ui/core/test-utils'
import Select from '@material-ui/core/Select'
import { DaySelect } from '../DaySelect'
import { startSetFirstDayOfTheWeek } from '../../../../data/settings/actions'

describe('<DaySelect />', () => {
  const mockStore = configureMockStore([thunk])

  let store
  let mount
  let wrapper

  beforeEach(() => {
    store = mockStore({ settings: { firstDayOfTheWeek: 'Monday' } })
    store.dispatch = jest.fn()
    mount = createMount()

    wrapper = mount(
      <Provider store={store}>
        <DaySelect />
      </Provider>
    )
  })

  test('should render <DaySelect /> correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('should dispatch action when onChange event is triggered', () => {
    wrapper.find(Select).prop('onChange')({ target: { value: 'Sunday' } })

    expect(store.dispatch).toHaveBeenCalled()
  })
})
