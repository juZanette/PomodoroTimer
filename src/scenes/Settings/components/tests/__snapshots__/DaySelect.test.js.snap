// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DaySelect /> should render <DaySelect /> correctly 1`] = `
<Styled(MuiBox)
  my={2}
>
  <Styled(WithStyles(ForwardRef(FormControl)))
    variant="outlined"
  >
    <WithStyles(ForwardRef(InputLabel))
      id="day-select"
    >
      Week starts on
    </WithStyles(ForwardRef(InputLabel))>
    <WithStyles(ForwardRef(Select))
      label="Week starts on"
      labelId="day-select"
      onChange={[Function]}
      value=""
    >
      <WithStyles(ForwardRef(MenuItem))
        value="Monday"
      >
        Monday
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        value="Sunday"
      >
        Sunday
      </WithStyles(ForwardRef(MenuItem))>
    </WithStyles(ForwardRef(Select))>
  </Styled(WithStyles(ForwardRef(FormControl)))>
</Styled(MuiBox)>
`;
